jsettlers.logic/src/jsettlers/logic/map/save/MapList.java

```diff
@@ -54,13 +50,13 @@
 	 * @return
 	 */
 	public static String getMapExtension() {
-		return CommonConstants.USE_SAVEGAME_COMPRESSION ? COMPRESSED_MAP_EXTENSION : MAP_EXTENSION;
+		return CommonConstants.USE_SAVEGAME_COMPRESSION ? MapLoader.MAP_EXTENSION_COMPRESSED : MapLoader.MAP_EXTENSION;
 	}
 
 	private static IMapListFactory mapListFactory = new IMapListFactory() {
 		@Override
-		public MapList getMapList() {
-			return new MapList(ResourceManager.getSaveDirectory());
+		public MapList getMapList(File originalSettlersFolder) {
+			return new MapList(ResourceManager.getSaveDirectory(), originalSettlersFolder);
```

Comment:
The ResourceManager should provide the original settlers folder. Or better: It should provide the folder containing the maps of the original game. Then it is most flexible and no fixed path/folder names need to be in this class.